### PR TITLE
Improve scalar literal format documentation

### PR DESCRIPTION
**Pull request by juliusv**
Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

**Comment by roidelapluie** [comment]
Thanks for pointing out this bug :)

#5984 should be applied to PromQL too :)

**Comment by roidelapluie** [comment]
Mmmh it seems like we are not affected by #5984 in promql. However, I suggest we note that we do not support these.

File: docs/querying/basics.md [hunk 1]
```diff
@@ -55,10 +55,19 @@ Example:
...
 ### Float literals
 
-Scalar float values can be literally written as numbers of the form
-`[-](digits)[.(digits)]`.
+Scalar float values can be literally written as integer or floating-point numbers, for example:
 
+    23
     -2.43
+    3.4e-9
+    0x8f
```
Comment (brian-brazil) [inline docs/querying/basics.md:63]: Is this something we actually support? I think we should stick to basic floats and ints.

File: docs/querying/basics.md [hunk 2]
```diff
@@ -55,10 +55,19 @@ Example:
...
+    -Inf
+
+Prometheus first attempts to parse scalar literals using Go's [`strconv.ParseInt()`](https://golang.org/pkg/strconv/#ParseInt)
```
Comment (brian-brazil) [inline docs/querying/basics.md:66]: I don't think we should be mentioning implementation details like this, also it's the PromQL grammar that defines this - not the Go implementation.

File: docs/querying/basics.md (hunk 1 above)
Comment (roidelapluie) [reply docs/querying/basics.md:63]: We do not support that and #5984 made sure we do not scrape that either.

Comment (roidelapluie) [reply docs/querying/basics.md:63]: Oh wait,yes we support that

Comment (brian-brazil) [reply docs/querying/basics.md:63]: Indeed, this is not something we support. https://github.com/prometheus/prometheus/blob/master/promql/parser/lex.go#L717

Comment (roidelapluie) [reply docs/querying/basics.md:63]: In practice: https://prometheus.demo.do.prometheus.io/graph?g0.range_input=1h&g0.expr=0x1&g0.tab=1

Comment (brian-brazil) [reply docs/querying/basics.md:63]: So it is

**Comment by juliusv** [comment]
It's funny how nobody was even sure what kind of number formats Prometheus accepts, good that we are documenting this :)

I removed the references to Go now and included a regex-based approach instead.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**